src/lib/users/dialogs/inst_user_first.rb

```diff
@@ -321,22 +392,27 @@ def process_new_user_form
         return false
       end
 
-      @password = pw1
-      return false unless valid_password?
-
       user.password = Y2Users::Password.create_plain(pw1)
-      root_user.password = Y2Users::Password.create_plain(pw1) if @use_pw_for_root
+
+      return false unless valid_password_for?(user)
+
+      update_config(user)
+
+      @password = pw1
 
       true
     end
 
-    # Registers the new user into {#users_config}
-    def attach_user(username, full_name: nil)
-      users_config.detach(user) if user
+    # Updates the users configuration with the given user
+    #
+    # @param user [Y2Users::User] the user for updating the configuration
+    def update_config(user)
+      main_user.name = user.name
+      main_user.gecos = user.gecos
+      main_user.password = user.password
+      root_user.password = user.password if @use_pw_for_root
```

Comment:
I'm not so happy with the naming here (well, not sure if I'm actually happy with the inst_user_first after restructure it again). I tried to avoid using `tmp_user` or `form_user` in the previous method (#process_new_user_form) and end up choosing `main_user` for the user who is going to receive the updates after validations succeeded. Improvements are welcome, as long as we decide to go ahead with this change (see https://github.com/yast/yast-users/pull/280#issuecomment-845570702)